### PR TITLE
new config setting for pnp backdata

### PR DESCRIPTION
**Pull request by AnneONS**
## Pull Request submission

An extra setting in the config will allow us to keep the BERD and PNP backdata coded in, to make it easy for when we switch between running the pipeline with the 2 different surveys.

In each case, though, the path to the backdata will be stored in config["staging_paths"]["backdata_path"]

The path_helpers function has been updated to facilitate this.

The pipeline should work up to the beginning of the Imputation module.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/utils/config.py [hunk 1]
```diff
@@ -314,7 +310,7 @@ def validate_freezing_run_config(config: dict) -> Tuple[bool, bool, bool, bool]:
         run_updates_and_freeze,
         run_with_frozen_data,
     ]
-    if len([val for val in values if val==True]) > 1:
```
Comment (AnneONS) [inline src/utils/config.py:317]: I changed this so the precommit would work as it doesn't like the ==True. The unit test still passes.

File: tests/test_utils/test_path_helpers.py [hunk 2]
```diff
@@ -146,6 +175,27 @@ def test_create_staging_config(config, expected_staging_dict):
...
+            "R:/DAP_emulation/2022_surveys/BERD/06_imputation/man_trim/trim_qa.csv"
+        ),
+
+        "backdata_path": "2021_data/backdata.csv",
+    }
+
+    staging_dict = create_staging_config(berd_config)
+
+    assert staging_dict == expected_staging_dict, "Staging config is not as expected"
+
```
Comment (AnneONS) [inline tests/test_utils/test_path_helpers.py:198]: New test to show that the staging config is correctly set up for BERD as well as PNP (previous test)

**Review by woodac — COMMENTED** [review]
Couldn't see any issues, logic is sound.

**Review by woodac — APPROVED** [review]
Can't see any issues with the code and the logic is sound. Happy to approve.

**Pull request merged**